### ISSUE TITLE
http://www.canalplus.fr

### ISSUE THREAD

**Issue by delaicre** [reek/anti-adblock-killer#3758]
The problem appears on videos.
http://www.canalplus.fr/cinema/emissions-cinema-sur-canal/pid6306-le-cercle.html?vid=1469930
https://imgur.com/a/w06Af
Firefox Quantum 57.0
Violentmonkey 
uBlock Origin

**Comment by uBlock-user** [comment]
@jspenguin2017 whitelisting `fwmrm` domain also pours in ads before videos.

**Comment by delaicre** [comment]
Sorry but it didn't change anything for me.

**Comment by jspenguin2017** [comment]
Hum, I had a solution for this domain, apparently it doesn't work anymore...

**Comment by jspenguin2017** [comment]
The loopback engine needed some tweak. The payload is still not quite right.

**Comment by jspenguin2017** [comment]
Actually, the video won't play either way, the website is broken.

**Comment by delaicre** [comment]
This is what I see without uBlock: https://imgur.com/a/HjZLM
It's the site of a french tv channel, did you get blocked by the geo-restriction?

**Comment by jspenguin2017** [comment]
Hum, let me try with a VPN

**Comment by jspenguin2017** [comment]
Oh yea, it works now, my solution still works then. 
![image](https://user-images.githubusercontent.com/7283682/33233188-676d0e06-d1cf-11e7-904d-d67e5d4e4629.png)

**Comment by jspenguin2017** [comment]
Oh yea, it works now, my solution still works then. 
![image](https://user-images.githubusercontent.com/7283682/33233188-676d0e06-d1cf-11e7-904d-d67e5d4e4629.png)

**Comment by delaicre** [comment]
Can you give me your solution, please? I wasn't here.

**Comment by jspenguin2017** [comment]
Only for Chrome. https://jspenguin2017.github.io/uBlockProtector/

**Comment by uBlock-user** [comment]
Is that an video ad in that picture ?

**Comment by delaicre** [comment]
> Is that an video ad in that picture ?

Nope but there is an idiotic one minute video add before the videos I want to see, the site doesn't have too much adds besides that.

**Comment by jspenguin2017** [comment]
I don't see any ads. And video works fine with current solution (if you have France VPN). 
```js
if (a.domCmp(["canalplus.fr"])) {
    a.loopbackXHR((ignored, url) => {
        if (url.includes(".v.fwmrm.net/ad/")) {
            return `<vmap:VMAP version='1.0' xmlns:vmap='http://www.iab.net/vmap-1.0'></vmap:VMAP>`;
        }
    });
}
```

**Comment by uBlock-user** [comment]
> Nope

It wasn't directed at you.

> I don't see any ads.

I meant it in the picture here -https://github.com/reek/anti-adblock-killer/issues/3758#issuecomment-346955867

Isn't that a video ad, atleast that's what it looks like it to me as it doesn't have video length info.

**Comment by delaicre** [comment]
Two adds actually.
https://imgur.com/a/cHE8i
> Add 1/2 : the video will start in 54 seconds.

I'm sure your solution is working well with Chrome jspenguin2017, I was talking about Firefox.
It's only one site, not a big problem. I was hoping to find a way but never mind.

**Comment by jspenguin2017** [comment]
@uBlock-user What do you mean? If you zoom in you'll see 56 min 05 sec on the right.

**Comment by uBlock-user** [comment]
Yep works fine without any whitelisting, that was the video and not ad, I confirmed it via fast forwarding the video and it's the entire video.

**Comment by jspenguin2017** [comment]
@delaicre 
The Quantum version is work in progress, although there isn't much progress recently due to a major design flaw in libmicro. 
libmicro is not needed when you are using uBO, so if you know how to sign extension packages then you can use this: https://github.com/AdBlockProtector2/AdBlockProtector2

**Comment by delaicre** [comment]
Thanks, I will try to make it work and look forward to the official release.
